app/dao/notifications_dao.py

```diff
@@ -64,10 +64,44 @@ def update_job_sent_count(notification):
 
 
 def dao_update_notification(notification):
+    notification.updated_at = datetime.utcnow()
     db.session.add(notification)
     db.session.commit()
 
 
+def update_notification_status_by_id(notification_id, status):
+    count = db.session.query(Notification).filter_by(
+        id=notification_id
+    ).update({
+        Notification.status: status,
+        Notification.updated_at: datetime.utcnow()
+    })
+    db.session.commit()
+    return count
+
+
+def update_notification_status_by_reference(reference, status):
+    count = db.session.query(Notification).filter_by(
+        reference=reference
+    ).update({
+        Notification.status: status,
+        Notification.updated_at: datetime.utcnow()
+    })
+    db.session.commit()
+    return count
+
+
+def update_notification_reference_by_id(id, reference):
+    count = db.session.query(Notification).filter_by(
+        id=id
+    ).update({
+        Notification.reference: reference,
+        Notification.updated_at: datetime.utcnow()
```

Comment:
The db model for notification uses an 'on update' clause to update the updated_at field.